crates/binjs_io/src/json/value.rs

```diff
@@ -0,0 +1,216 @@
+/// JS value in JSON.
+use TokenReaderError;
+
+use binjs_shared::SharedString;
+
+use std::borrow::Borrow;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+/// JS value in JSON.
+pub enum Value {
+    String(SharedString),
+    Number(f64),
+    Object(Rc<HashMap<SharedString, Rc<Value>>>),
```

Comment:
You should probably define
```rust
pub type Object = Rc<HashMap<SharedString, Rc<Value>>>;
```

This would make some things easier to read.

Reply:
@arai-a It's my understanding that we don't intend to use this module for complicated code, only for generating test files, right?
If so, we don't care all that much about performance.

Reply:
Isn't this going to be used for regular parsing of input from Shift? But also, if we don't care about performance, then I see even less reasons to use `Rc` here.

Reply:
@arai-a, @RReverser and myself have different understanding of the objective of this code. Could you settle the issue? :)

Reply:
`Rc` there is just to make it easier to have reference to a certain object in the tree while reading (traversing the tree), and I cannot find another way...
If there's another simpler way without `Rc`, I'm happy to use it.

then, at this point this is only for a testing-purpose-only command.
The implementation will be replaced with serde-json? (@RReverser would know better about the plan I think?)